units/check_blocked_tasks.go

```diff
@@ -135,21 +139,21 @@ func checkUnmarkedBlockingTasks(ctx context.Context, t *task.Task, dependencyCac
 		return nil
 	}
 
-	blockingTasks, err := t.RefreshBlockedDependencies(dependencyCaches)
+	finishedBlockingTasks, err := t.GetFinishedBlockingDependencies(dependencyCaches)
 	catcher.Wrap(err, "getting blocking tasks")
 	blockingTaskIds := []string{}
 	if err == nil {
-		for _, blockingTask := range blockingTasks {
+		for _, blockingTask := range finishedBlockingTasks {
```

Comment:
Re this [comment](https://github.com/evergreen-ci/evergreen/pull/6900#discussion_r1320279512): should we include an extra `MarkDependenciesFinished` call here to updates all direct dependencies on this task to cache whether or not this task has finished running